### ISSUE TITLE
[PEP 728] "closed" `TypedDict` with `__extra_items__` cannot be type-checked correctly as an `TypedDict` attribute

### ISSUE THREAD

**Issue by stevapple** [microsoft/pyright#8503]
**Describe the bug**
When using [PEP 728](https://peps.python.org/pep-0728/) (through `enableExperimentalFeatures`), we found using a "closed" `TypedDict` with `__extra_items__` as the type of attribute in another `TypedDict` will cause unexpected behaviors:
- It will falsely reject valid literal inputs as `__extra_items__`;
- It will falsely accept invalid inputs as regular attributes.

**Code or Screenshots**

With `typing_extensions` 4.12.2 and Python 3.12.4:

```python
from typing_extensions import TypedDict

class ExtraInfo(TypedDict, closed=True):
    __extra_items__: str | int

class ExtensibleInfo(TypedDict, closed=True):
    # eg. name: str
    __extra_items__: str | int | ExtraInfo

class FixedInfo(TypedDict):
    # eg. name: str
    extra: ExtraInfo

# false error: `dict[str, int]` should be compatible with `ExtraInfo`
extensibleInfo: ExtensibleInfo = {
    "birth": {
        "year": 2000,
        "month": 12,
        "day": 31,
    }
}

# false ok: `ExtraInfo` doesn't allow `bool` values
fixedInfo: FixedInfo = {
    "extra": {
        "birth.year": 2000,
        "birth.month": 12,
        "birth.day": 31,
        "graduated": False,
    }
}
```

**VS Code extension or command-line**
```
[info] (Client) Pylance async client (2024.7.1) started with python extension (2024.10.0)
[info] [Info  - 15:41:34] (13212) Pylance language server 2024.7.1 (pyright version 1.1.369, commit 1168be71) starting
```

**Comment by erictraut** [comment]
Thanks for the bug report. Pyright's current implementation doesn't support bidirectional type inference for `__extra_items__`, but it's reasonable to expect that it should. I've added that support, and it will appear in the next release. 

The second issue you reported (the false negative) isn't a bug because `bool` is a subtype of `int`. Therefore it's not a type violation to include a value of `False`. If you modify your example and replace `False` with `[]`, you'll see that pyright does detect the type violation.

I'll note that PEP 728 has not yet been accepted, which explains why this is considered "experimental support". I recommend avoiding this mechanism in production code. If the PEP is ultimately rejected, this support will be removed from pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.373.
